### PR TITLE
Display release notes directly in DevTools

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Fixes https://github.com/flutter/devtools/issues/3291.

The workflow is as such:
- check in the local `.flutter-devtools/.devtools` file for the last version of release notes we showed in DevTools.
- compare this version to the current DevTools version
- if the current DevTools version is newer, try to pull release notes for this version from the flutter website
- if the flutter website has release notes for this version, display them in devtools
- add an entry to the local `.flutter-devtools/.devtools` file for the newest version of release notes we have shown. 

embedded-release-notes

CC @InMatrix @cobblest 

NOTE: this cannot be landed in the current state. We are hitting a CORS exception when trying to access the devtools release notes resources from flutter.dev @johnpryan

### CODE REVIEW

File: packages/devtools_app/lib/src/release_notes.dart [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  ValueListenable<String> get releaseNotesMarkdown => _releaseNotesMarkdown;
+
+  final _releaseNotesMarkdown = ValueNotifier<String>(null);
+
+  ValueListenable<bool> get releaseNotesVisible => _releaseNotesVisible;
+
+  final _releaseNotesVisible = ValueNotifier<bool>(false);
+
+  void _init() {
+    if (server.isDevToolsServerAvailable && !isEmbedded()) {
```
Comment (elliette) [inline packages/devtools_app/lib/src/release_notes.dart:169]: OOC, when is `isEmbedded` true?

File: packages/devtools_app/lib/src/release_notes.dart [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+      try {
+        String releaseNotesMarkdown =
+            await http.read(Uri.parse(_releaseNotesUrl(devtoolsVersion)));
+        // This is a workaround so that the images in release notes will appear.
+        // The {{site.url}} syntax is best practices for the flutter website
+        // repo, where these release notes are hosted, so we are performing this
+        // workaround on our end to ensure the images render properly.
+        releaseNotesMarkdown = releaseNotesMarkdown.replaceAll(
+          '{{site.url}}',
+          'https://docs.flutter.dev',
```
Comment (elliette) [inline packages/devtools_app/lib/src/release_notes.dart:195]: Suggestion: Change this and path in `_releaseNotesUrl` to top-level variables

**Review by elliette — APPROVED** [review]

**Comment by johnpryan** [comment]
@kenzieschmoll does https://github.com/flutter/website/pull/6579 fix the CORS exception?

**Comment by kenzieschmoll** [comment]
> does flutter/website#6579 fix the CORS exception?

Just verified this - yes it does - thanks!

File: packages/devtools_app/lib/src/release_notes.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/release_notes.dart:169]: isEmbedded is true when DevTools is displayed inside the IDE tooling windows

File: packages/devtools_app/lib/src/release_notes.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/release_notes.dart]: done

**Pull request merged**